### DISCUSSION TITLE
Search of repo returns no code results

### DISCUSSION THREAD

**Discussion by mhilbush** [community/community#44388] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Doing a search of the following repo returns no code results. Commits, issues, etc are returned. Just no code results. It worked previously, but I noticed a week ago it was no longer returning code results.

https://github.com/openhab/openhab-addons

The following search should return hundreds of code hits, yet there are zero hits.

https://github.com/openhab/openhab-addons/search?q=BaseThingHandler

**Comment by rashidwassan** [comment]
It is working on my end. Go to the commits section and you can see commits with messages containing the search query.

**Reply by rashidwassan** [reply]
Only commits are returned.

**Reply by wborn** [reply]
When the same code is added to a new repo it is possible to search the code so it looks like some kind of indexing problem.

E.g. I created a new repo (not a fork) with the same code which does return the expected results:

https://github.com/wborn/openhab-addons-search/search?q=BaseThingHandler

Still no results for the same search in openhab/openhab-addons:

https://github.com/openhab/openhab-addons/search?q=BaseThingHandler

**Comment by mhilbush** [comment]
> It is working on my end.

Sorry, what do you mean by "working"? Are you saying that when you execute the following search that you see **code** results?

https://github.com/openhab/openhab-addons/search?q=BaseThingHandler

Here's what I see.
<img width="917" alt="repo-search-results" src="https://user-images.githubusercontent.com/19286354/212693674-295cc9f7-5c6e-44a0-b506-b906f6cf637d.png">

There should be ~397 code results.

**Comment by luizfm** [comment]
I'm facing a similar issue when trying to search for issues with a user + repo. This is the API url I'm trying to get issues information:

https://api.github.com/search/issues?q=A%20repo:luizfm/github-blog-reactjs

It returns `items: []`, when it should have at least 1 issue. I've tried to close one and create a new one as well and no updates happen on the results of this endpoint.

Am I doing the wrong query here?

<img width="856" alt="Screen Shot 2023-01-16 at 5 56 05 PM" src="https://user-images.githubusercontent.com/45155140/212764394-c330ee32-d68e-4b51-8201-3c2343448e0d.png">

This is the repo url https://github.com/luizfm/github-blog-reactjs.

Btw, I'm commenting on this post because it seems to be a general issue with the `Search API` that is not affecting only the `codes` results but also other topics results.

**Reply by luizfm** [reply]
[Solved]. The way here was turn my repo private and public again and it seems that it removed some kind of cached information.

**Reply by lsiepel** [reply]
Are you aware if there is another way to force/refresh as togling private/public seems to detach all forks and they probably don;t re-attach automatically.

**Reply by luizfm** [reply]
Hummm, not pretty sure if there is any other option. If so, that would be nice... For me is not a problem at all, since I was playing around with this API to complete a small course challenge 😬, but for anyone else that has this issue on a lib repo it might be a problem.

**Reply by mhilbush** [reply]
The only other option I can see is to join the new "Code Search and Code View" beta, as it doesn't have any problem searching the repo. I'll assume that the new search uses a different indexing scheme, which might explain why it works.  It seems to take only a day or so to get added to the beta.

**Comment by jasonrohrer** [comment]
Same bug for me, but only on certain repositories, like PHP:

https://github.com/php/php-src/search?q=http_response_code

This returns 0 Code results (even though that string certainly appears in the code).

Whereas, on other repos, code search is working:

https://github.com/jasonrohrer/minorGems/search?q=SimpleVector

This returns 69 Code results, which sounds about right.

Maybe it has something to do with the size of the repo?  MinorGems is tiny, where PHP is big.

**Comment by mhilbush** [comment]
> Maybe it has something to do with the size of the repo?

I don't think it has anything to do with the size. There are repos out there that are way bigger than https://github.com/openhab/openhab-addons where search works just fine.

BTW, I joined GitHub's new "Code Search and Code View" beta. Searching https://github.com/openhab/openhab-addons and https://github.com/php/php-src/search?q=http_response_code works just fine. I suspect the new "Code Search and Code View" uses different indexes (and different indexing tech for that matter).

Another data point is that you can place the contents of https://github.com/openhab/openhab-addons into a new repo (not a fork) and searching works fine.

Tends to make me believe there's index corruption that needs to be repaired/rebuilt. But I have no idea how to do that, or who to ask to do it.

**Comment by mhilbush** [comment, marked as answer]
Joining the new "Code Search and Code View" beta resolved the problem for me, as the new search doesn't have any problem searching the repo. It seems to take only a day or so to get added to the beta.

**Reply by karl-police** [reply]
> Joining the new "Code Search and Code View" beta resolved the problem for me, as the new search doesn't have any problem searching the repo. It seems to take only a day or so to get added to the beta.

I'd only want to join the Code Search but not the Code View.

**Reply by snowman** [reply]
NOT work either.

Try https://github.com/search?q=repo:hoothin/UserScripts+i18nData&type=code

![image](https://user-images.githubusercontent.com/16096559/225223367-aff44d66-f9b5-45aa-84ef-477648f1d0b5.png)

![image](https://user-images.githubusercontent.com/16096559/225223294-49d3754f-86d9-4aa2-b2f3-fd7205ab3112.png)

**Comment by astratagem** [comment]
FWIW, I had been using the new code search feature but stopped for similar reasons – even using the browser's built-in search functionality failed to find results in a page even though I knew they were there. And I thought it couldn't get any worse, but here we are with the "stable" search functionality broken to an even more severe degree…

<img width="886" alt="Screen Shot 2023-01-29 at 9 55 27 PM" src="https://user-images.githubusercontent.com/1757914/215378908-a5aef5ee-1127-421f-967c-3a11159b1d2c.png">

**Comment by damien-johnston** [comment]
Same issue here, I posted in community.

https://github.com/orgs/community/discussions/38692#discussioncomment-5027770

**Reply by damien-johnston** [reply]
Figured out a work around for ict-index search on GitHub, tested with a GH user that doesn't have the beta search enabled.

Here's how it works.
Login to GitHub with your user.
https://github.com/

Then go to the old search URL. cs is the old code search and it still works!
https://cs.github.com/

Any search you do must start with this term to filter it to the ict index repo.
repo:damien-johnston/ict-index

Here's an example search for the term "fair value gap" and "NWOG"
repo:damien-johnston/ict-index "fair value gap" AND "NWOG"

I like the old UI results better tbh.
![image](https://user-images.githubusercontent.com/13722748/220470064-c0bbc233-682d-42ec-94c3-e6e3bbead2c9.png)

![image](https://user-images.githubusercontent.com/13722748/220470108-327e5ed4-3913-476a-86e0-656a95852fe4.png)

**Reply by nikescar1** [reply]
Am I doing it wrong? I try to go to the old search and it redirects me to https://github.com/features/code-search

**Reply by damien-johnston** [reply]
Nope looks like there is a caveat, and or seems there has been some sort of fix. If your user has never signed up for beta search it will be redirected. The user I tested with had the feature disabled.

 
This is a different user that has never signed up for the beta search.
![image](https://user-images.githubusercontent.com/13722748/220764829-a5ab3568-df89-4988-8e56-ba20a4293f8f.png)

I can now search my existing repo without going to the cs.github url. with a user that has not signed up for the beta search. This didn't work a few days ago.

Searching for some text.
![image](https://user-images.githubusercontent.com/13722748/220766465-b1638c26-fcb7-4d73-97a0-386cbb5e2d68.png)

Results from search without beta search preview.
![image](https://user-images.githubusercontent.com/13722748/220766996-046fc5ba-879f-4d08-bc9f-0454194af658.png)

hth

**Comment by dashesy** [comment]
Just try searching anything here: https://github.com/hwchase17/langchain no code results! Search is broken

**Comment by johnddx** [comment]
No results returned for code at https://github.com/videolan/vlc either.

**Comment by maziarzamani** [comment]
I have similar issues with private repos.

**Comment by kcrane3576** [comment]
Experiencing the same issue as well for Search API with many public repos (200 response for all).

**Comment by airtonix** [comment]
Pretty much all our Private repos are not returning search results.
